src/fragment-generation-utils.js

```diff
@@ -39,9 +68,15 @@ export const GenerateFragmentStatus = {
  * URL, which will highlight the given selection upon opening.
  * @param {Selection} selection - a Selection object, the result of
  *     window.getSelection
+ * @param {Date} [startTime] - the time when generation began, for timeout
+ *     purposes. Defaults to current timestamp.
  * @return {GenerateFragmentResult}
+ * @throws {Error} - Will throw if computation takes longer than the accepted
```

Comment:
Is the plan to add a try-catch in the calling JS?

Reply:
Yup!